docs/markdown/Using Pants/using-pants-in-ci.md

```diff
@@ -15,15 +15,46 @@ Directories to cache
 
 In your CI's config file, we recommend caching these directories:
 
-- `$HOME/.cache/pants/setup`: the initial bootstrapping of Pants.
-- `$HOME/.cache/pants/named_caches`: caches of tools like pip and PEX.
-- `$HOME/.cache/pants/lmdb_store`: cached content for prior Pants runs, e.g. prior test results.
+
+- `$HOME/.cache/pants/setup`<br>
+  This is the Pants bootstrap directory. Cache this against the version, as specified
+  in `pants.toml`.  E.g., if using GitHub Actions you can use:
+  ```yaml
+  path: |
+    ~/.cache/pants/setup
+  key: pants-setup-${{ runner.os }}-${{ hashFiles('pants.toml') }}
+  restore-keys: |
+    pants-setup-${{ runner.os }}-${{ hashFiles('pants.toml') }}
+    pants-setup-${{ runner.os }}-
+  ```
+- `$HOME/.cache/pants/named_caches`<br>
+  Caches used by some underlying tools.  Cache this against the inputs to those tools.
+  For the `pants.backend.python` backend, named caches are used by PEX, and therefore
+  its inputs are your lockfiles. For example:
+  ```yaml
+  path: |
+    ~/.cache/pants/named_caches
+  key: named-caches-${{ runner.os }}-${{ hashFiles('pants.toml') }}-${{ hashFiles('python-default.lock') }}
+  restore-keys: |
+    named-caches-${{ runner.os }}-${{ hashFiles('pants.toml') }}-${{ hashFiles('python-default.lock') }}
+    named-caches-${{ runner.os }}-${{ hashFiles('pants.toml') }}-
+    named-caches-${{ runner.os }}-
+  ```
+
+If you're not using a fine-grained [remote caching](doc:remote-caching-execution) service,
+then you may also want to preserve the local Pants cache at `$HOME/.cache/pants/lmdb_store`.
+This has to be invalidated on any file that can affect any process, e.g., `hashFiles('**/*')`.
+Computing such a coarse hash, and saving and restoring large directories, can be unwieldy.
+So this may be impractical and slow on medium and large repos.
+A [remote cache service](doc:remote-caching-execution) integrates with Pants's fine-grained
+invalidation and avoids these problems, and is recommended for the best CI performance.
```

Comment:
Another way to skin this cat that I don't see being discussed it self-hosted GitHub Runners, which can leverage a persistent filesystem.

https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

Reply:
I have zero experience doing that, and again, it is out of scope for this change I think. We couldn't document that until we tried it.

Reply:
I'll file an issue and we can document later

Reply:
Is this something you've actually done? It's a really interesting approach if so!